### ISSUE TITLE
Question about the segmentation loss

### ISSUE THREAD

**Issue by valeriopaolicelli** [HanjiangHu/DASGIL#5]
Hi, I would like to have some explanations about the seg. loss. I noticed that in your paper you wrote 'Similar to the image forward propagation for depth reconstruction, we apply score regression and the softmax layer for the outputs **at multiple layers** of generator Gs'. 
Instead, analyzing your code in backward_G function of model.py, the seg loss seems to be applied just on the last output of Gs (lines 271-273), differently from the depth loss which is applied to all outputs from the 5th layer of Gd as reported in the paper. 
Am I missing something?
Thanks

**Comment by HanjiangHu** [comment]
Thanks a lot for your interest in our work. Yes, we finally use the last layer of Gs instead of the multi-layer setting to optimize the performance and architecture, which has been clarified in our latest arxiv version of the [paper](https://arxiv.org/pdf/2010.00573.pdf) and the published version in [TIP](https://ieeexplore.ieee.org/document/9296559). Some details are also added or clarified based on some comments of reviewers in the revised version. Hope to be helpful to you.
Thanks
Hanjiang

**Comment by valeriopaolicelli** [comment]
Thanks for the fast reply. I read the old version of your paper, so now I update it with the one linked above.
I have few questions regarding the results tables (I, II and III). What kind on training data and network implementations did you use? For instance, what is the NetVLAD encoder (alexnet or vgg16) and what are the training data used for it? Just virtual kitty?
Thanks a lot for the support.

**Comment by HanjiangHu** [comment]
Thanks for your question. For the baseline results in tables, we directly use the results from the official benchmark [website](https://www.visuallocalization.net/benchmark/) and some of the details of these baselines can be found in the benchmark [paper](https://openaccess.thecvf.com/content_cvpr_2018/papers/Sattler_Benchmarking_6DOF_Outdoor_CVPR_2018_paper.pdf). These results are with the same test set of CMU-Seasons dataset but might with different training sets. Hope to dismiss your concern.
Hanjiang

**Comment by valeriopaolicelli** [comment]
Thanks for your support. Will you also upload the code and the dataset splits to test place recognition on RobotCar?

**Comment by HanjiangHu** [comment]
We follow the split set of Oxford RobotCar dataset in the previous work of [ICRA2019](https://ieeexplore.ieee.org/abstract/document/8794221) and recent [IJCV](https://link.springer.com/article/10.1007/s11263-020-01363-6). Currently, we are not considering making this experiment code and dataset splits public but it might be available in the future.
Thanks
Hanjiang
